### PR TITLE
DDEV 1.19 - Disable seldom used docker services by default. Change DB override

### PR DESCRIPTION
**Pull request by weitzman**
Saves battery life.

**Description:**
Introduces a new .env file in order to override DB image for non-super sanitized or for M1 Mac. 

Lets hold this until DDEV 1.19 is released. Then we merge this and ask everyone to upgrade their DDEV.

**Jira:** (Skip unless you are MA staff)
DP-24231

**To Test:**
- [x] Upgrade DDEV and start the project. If it starts and works we have done no harm. Copy the new .ddev/.env.example file to .ddev/.env and specify a non-default image. Then restart DDEV and assure you got the specified image. I used our Portainer service to verify this - http://mass.local:8089. Search for the Container `ddev-mass-dbmass` and scroll down - see image name under _Container details_.
- [ ] Start up and assume that backstop and chromedriver are not listed in `ddev describe` output. Then run ddev service enable backstop` and `ddev restart`. Assure that backstop now shows up in `ddev describe` output.

**After merge**
Announce in SSR channel:
- DDEV 1.19 is now required. Right after you do your next git pull, please upgrade your DDEV (`brew upgrade ddev`)and move your `.env` file to `.ddev/.env`. Then `ddev poweroff && ddev start`. More info at https://github.com/massgov/openmass/pull/1388

---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)

### CODE REVIEW

File: .ddev/config.yaml [hunk 1]
```diff
@@ -3,7 +3,6 @@ type: drupal9
 docroot: docroot
 php_version: "7.4"
 webserver_type: apache-fpm
-dbimage: massgov/mysql-sanitized:super
```
Comment (deviantintegral) [inline .ddev/config.yaml:6]: Do we know if this is an unintended compatibility break in ddev, or is there something "wrong" with how this was done before?

Comment (weitzman) [reply .ddev/config.yaml:6]: We were relying on ddev internals. I worked with rfay on this new approach.

File: .ddev/.disabled-services/docker-compose.chromedriver.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+---
+# Docker-ChromeDriver configuration for DDEV-Local.
```
Comment (deviantintegral) [inline .ddev/.disabled-services/docker-compose.chromedriver.yaml]: Off topic, but seeing this you may be interested in https://github.com/lullabot/drainpipe and https://github.com/lullabot/drainpipe-dev.

File: README.md [hunk 3]
```diff
@@ -49,9 +49,10 @@ If the Docker section above is unappealing, its easy to run mass.gov natively on
 ###### Notes
 - The site is browseable at https://mass.local
 - It takes a few minutes for the `dbmass` container start up.
-- [You may override ddev config locally](https://ddev.readthedocs.io/en/stable/users/extend/config_yaml/). For example, create a `.ddev/config.personal.yml` file and add a line `dbimage: massgov/mysql-sanitized:latest` to use our less sanitized DB.
-- Similarly, rename [.ddev/config.arm.yaml.example](https://github.com/massgov/openmass/blob/develop/.ddev/config.arm.yaml.example) to config.arm.yaml in order to use ARM containers suitable for the Apple M1 Macs.
+- [You may override ddev config locally](https://ddev.readthedocs.io/en/stable/users/extend/config_yaml/). create a `.ddev/config.personal.yml` file and add whatever you need.
+- Similarly, rename [.ddev/.env.example](https://github.com/massgov/openmass/blob/develop/.ddev/.env.example) to `.env` in order to use ARM containers suitable for the Apple M1 Macs. This is also how you specify the less sanitized variant of our database.
```
Comment (deviantintegral) [inline README.md:53]: Another option would be to publish the containers with the same name? Then docker should default to the currently running platform unless `--platform` is specified.

Home Assistant requires putting the architecture name into the image name (I think this pattern predates proper multi-platform support in docker), and it's a real pain with it being coded that way everywhere.

```console
docker run --rm -it --platform linux/arm64 alpine:latest uname -a
Linux cacc67861cb1 5.10.88-0-virt #1-Alpine SMP Tue, 28 Dec 2021 12:51:14 +0000 aarch64 Linux

docker run --rm -it --platform linux/amd64 alpine:latest uname -a
Unable to find image 'alpine:latest' locally
latest: Pulling from library/alpine
Digest: sha256:21a3deaa0d32a8057914f36584b5288d2e5ecc984380bc0118285c70fa8c9300
Status: Downloaded newer image for alpine:latest
Linux 2ab351d5684e 5.10.88-0-virt #1-Alpine SMP Tue, 28 Dec 2021 12:51:14 +0000 x86_64 Linux

File: .ddev/.disabled-services/docker-compose.chromedriver.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+# There is one item that must be configured in this file in order for it to
+# work, please see below.
+
+version: '3.6'
+services:
+  chromedriver:
+    image:  previousnext/chrome-headless:65
```
Comment (deviantintegral) [inline .ddev/.disabled-services/docker-compose.chromedriver.yaml]: Is this really referring to Chrome 65? I don't see a repo for https://hub.docker.com/r/previousnext/chrome-headless/tags.

... looks like it:

```console
docker run --rm -it previousnext/chrome-headless:65 /opt/google/google-chrome --version
Google Chrome 65.0.3325.162
```

**Review by deviantintegral — COMMENTED** [review]
- [x] Able to run ddev start and load the site.
- [ ] Haven't tested the env file yet.
- [ ] Looks like backstop and chromedriver are still running

I wonder if there's stale configs. I now see this file as not in the git tree. I can investigate further later:

```console
On branch DP-24231_ddev-1.19
Your branch is up to date with 'origin/DP-24231_ddev-1.19'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	../.ddev/docker-compose.backstop.yaml
```

**Comment by deviantintegral** [comment]
I found the chromedriver issue; `.ddev/docker-compose.chromedriver.yaml` is still in the repo, so I think it needs to be removed.

As well, developers may need to remove untracked files after merging, since they won't be removed due to our gitignore rules.

File: README.md (hunk 3 above)
Comment (weitzman) [reply README.md:53]: I looked at docs for making a multi-platform image and wasnt able to get very far. Happy to discuss that more.

File: .ddev/.disabled-services/docker-compose.chromedriver.yaml (hunk 2 above)
Comment (weitzman) [reply .ddev/.disabled-services/docker-compose.chromedriver.yaml]: For this project I'd be interested in the composer-lock-diff integration.

File: .ddev/.disabled-services/docker-compose.chromedriver.yaml (hunk 4 above)
Comment (weitzman) [reply .ddev/.disabled-services/docker-compose.chromedriver.yaml]: I'm happy to get off of that image. I. tried once and couldn't get to green.

Comment (deviantintegral) [reply .ddev/.disabled-services/docker-compose.chromedriver.yaml]: I filed https://massgov.atlassian.net/browse/DP-24306 for this given Chrome is up to version 99.

**Review by deviantintegral — APPROVED** [review]
The database changes look good to me.

I was a little confused that the testing steps didn't mention enabling the chromedriver service. Is it likely that team members are using Backstop, but pointing to chrome on their host?

Should we be gitignore'ing `.ddev/docker-compose.backstop.yaml` and `.ddev/docker-compose.chromedriver.yaml` (which are created when running `ddev service enable`)?

ddev 1.19 is now stable. If I'm right about ignoring those two files, let's add that in and merge. Otherwise this looks good to me! I'm so happy I won't have to manually symlink the colima docker socket after rebooting.

**Comment by deviantintegral** [comment]
I marked as ready for review given ddev 1.19 is now stable.

**Comment by weitzman** [comment]
Pushed those changes to .gitignore.

Chromedriver service is used by some of our DTT tests. I don't think Backstop uses that service. Backstop has a headless Chrome inside of its own docker image, I think.

**Pull request merged**